aiida/common/lang.py

```diff
@@ -14,6 +14,49 @@
 
 import functools
 import inspect
+import keyword
+import six
+
+# Python 3 has a nice built-in solution, but while we support python 2 we need this ugly switch
+if six.PY2:
+
+    def isidentifier(identifier):
+        """Return whether the given string is a valid python identifier.
+
+        :return: boolean, True if identifier is valid, False otherwise
+        :raises TypeError: if identifier is not string type
+        """
+        import string
+
+        type_check(identifier, six.string_types)
+
+        if not identifier:
+            return False
+
+        if keyword.iskeyword(identifier):
+            return False
+
+        first = '_' + string.lowercase + string.uppercase  # pylint: disable=no-member
+        if identifier[0] not in first:
+            return False
+
+        other = first + string.digits
+        for character in identifier[1:]:
+            if character not in other:
+                return False
+
+        return True
+
+else:
+
+    def isidentifier(identifier):
+        """Return whether the given string is a valid python identifier.
+
+        :return: boolean, True if identifier is valid, False otherwise
+        :raises TypeError: if identifier is not string type
+        """
+        type_check(identifier, six.string_types)
+        return identifier.isidentifier() and not keyword.iskeyword(identifier)
```

Comment:
Just as a note (maybe to comment, to avoid to forget) - there is a difference between the PY2 and PY3 implementations - the latter allows unicode characters. Tests pass because below in `validate_port_name` you also check the valid character set.

Reply:
But certain unicode characters *are* valid in identifiers, no?

Reply:
I think: only ASCII are valid in py2, Unicode ([with rules](https://www.python.org/dev/peps/pep-3131/)) is allowed in py3, and in AiiDA we don't want Unicode at this stage, I think. The implementation of this PR is right, my only point here is that the py2 and py3 implementations here do different things on the same string (although they both do the "correct thing" of telling if it's an identifier in the current version of the language). A comment to clarify this would be enough I think.

Reply:
I see, will do